public/app/features/correlations/components/Table/index.tsx

```diff
@@ -0,0 +1,161 @@
+import { cx, css } from '@emotion/css';
+import React, { useMemo, Fragment, ReactNode } from 'react';
+import {
+  CellProps,
+  SortByFn,
+  useExpanded,
+  useSortBy,
+  useTable,
+  DefaultSortTypes,
+  TableOptions,
+  IdType,
+} from 'react-table';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Icon, useStyles2 } from '@grafana/ui';
+import { isTruthy } from 'app/core/utils/types';
+
+import { EXPANDER_CELL_ID, getColumns } from './utils';
+
+const getStyles = (theme: GrafanaTheme2) => ({
+  table: css`
+    border-radius: ${theme.shape.borderRadius()};
+    border: solid 1px ${theme.colors.border.weak};
+    background-color: ${theme.colors.background.secondary};
+    width: 100%;
+    td,
+    th {
+      padding: ${theme.spacing(1)};
+      min-width: ${theme.spacing(3)};
+    }
+  `,
+  evenRow: css`
+    background: ${theme.colors.background.primary};
+  `,
+  shrink: css`
+    width: 0%;
+  `,
+});
+
+export interface Column<TableData extends object> {
+  /**
+   * ID of the column.
+   * Set this to the matching object key of your data or `undefined` if the column doesn't have any associated data with it.
+   * This must be unique among all other columns.
+   */
+  id?: IdType<TableData>;
+  cell?: (props: CellProps<TableData>) => ReactNode;
+  header?: (() => ReactNode | string) | string;
+  sortType?: DefaultSortTypes | SortByFn<TableData>;
+  shrink?: boolean;
+  visible?: (col: TableData[]) => boolean;
+}
+
+interface Props<TableData extends object> {
+  columns: Array<Column<TableData>>;
+  data: TableData[];
+  renderExpandedRow?: (row: TableData) => JSX.Element;
+  className?: string;
+  getRowId: TableOptions<TableData>['getRowId'];
+}
+
+/**
+ * non-viz table component.
+ * Will need most likely to be moved in @grafana/ui
+ */
```

Comment:
Why aren't you adding it there in this PR? In case this isn't possible right now are you going to open an issue for it?

Reply:
because this is still in WIP, we are working with the UX  team and on a design doc right now to define what needs to be done for it to be in the ui package. there's already a draft issue in our board for this.